src/main/java/wolox/training/controller/BookController.java

```diff
@@ -0,0 +1,80 @@
+package wolox.training.controller;
+
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.execptions.BookException;
+import wolox.training.execptions.BookExistException;
+import wolox.training.execptions.BookIdMismatchException;
+import wolox.training.execptions.BookNotFoundException;
+import wolox.training.model.Book;
+import wolox.training.repository.BookRepository;
+
+@RestController
+@RequestMapping("/api/books")
+public class BookController {
+
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookController(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
+
+    @GetMapping
+    @ResponseStatus(HttpStatus.OK)
+    public List<Book> findAll() {
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/author")
+    public ResponseEntity<Book> findOneByAuthor(@RequestParam(name = "author") String author) {
+        return ResponseEntity.ok(bookRepository.findFirstByAuthor(author)
+                .orElseThrow(BookNotFoundException::new));
+    }
+
+    @PostMapping
+    public ResponseEntity<Book> create(@RequestBody Book book) {
+        if (book.getId() != null) {
```

Comment:
Here you only check for the id value en the body, not the existence in database. If the id is sent in the body at creation, it should be a bad request error. If there is any unique key sent in the body, you could check it uniqueness against the database.

Reply:
I see now that you throw a Bad Request error with this. Maybe it is a little confusing saying already exists just because of a parameter in the body.